### ISSUE TITLE
Random baseline treats DSR reward as log return: np.exp(rewards)-1 invalid for differential_sharpe

### ISSUE THREAD

**Issue by kwojdalski** [kwojdalski/masters_thesis#49]
**File:** src/trading_rl/evaluation/statistical_benchmarks.py:200-202
**Category:** 5 — Benchmark and Metric Errors
**Severity:** MEDIUM

**What it computes:**
```python
rewards = (
    rollout["next", "reward"].detach().cpu().reshape(-1).numpy()[:max_steps]
)
random_returns.append(np.exp(rewards) - 1.0)
```
`np.exp(reward) - 1` is the correct formula to convert log returns to simple returns — but only when `reward_type == "log_return"`. For `reward_type == "differential_sharpe"`, the reward at each step is a DSR value (typically in the range ±10 after clipping). `np.exp(10.0) - 1 ≈ 22026`, which is not a simple return.

**What it should compute:**
For `log_return` reward type: the current formula is correct.
For `differential_sharpe` (and any non-log-return reward), the random baseline returns should be extracted from the environment's NLV path — the same way `compute_strategy_simple_returns_for_split` handles it — not from the reward stream.

**Impact:** When `reward_type == "differential_sharpe"` and `compare_to_random=True`, the random baseline "returns" are orders-of-magnitude larger than the strategy's actual NLV-based returns. Statistical tests (t-test, Sharpe bootstrap, etc.) compare these incommensurable series, producing completely invalid p-values and confidence intervals. The random baseline will always appear dramatically different from the strategy, not because the strategy is good, but because the units are wrong.

**Proposed fix:**
Add a reward-type check:
```python
# statistical_benchmarks.py, inside compute_random_baseline_returns loop
from trading_rl.evaluation.returns import extract_tradingenv_returns
from trading_rl.constants import RewardType

if reward_type == RewardType.LOG_RETURN:
    rewards = rollout["next", "reward"].detach().cpu().reshape(-1).numpy()[:max_steps]
    random_returns.append(np.exp(rewards) - 1.0)
else:
    # Extract actual NLV-based returns for non-log-return reward types
    cumulative = extract_tradingenv_returns(env, max_steps)
    if cumulative is not None and len(cumulative) > 1:
        step_log = np.diff(cumulative)
        random_returns.append(np.exp(step_log) - 1.0)
    # else: skip or fall back
```
The function signature needs a `reward_type` parameter added.
